rpc/lib/client/http_client.go

```diff
@@ -195,7 +206,15 @@ func (c *JSONRPCClient) Call(method string, params map[string]interface{}, resul
 		return nil, err
 	}
 	requestBuf := bytes.NewBuffer(requestBytes)
-	httpResponse, err := c.client.Post(c.address, "text/json", requestBuf)
+	httpRequest, err := http.NewRequest(http.MethodPost, c.address, requestBuf)
+	if err != nil {
+		return nil, err
+	}
+	httpRequest.Header.Set("Content-Type", "text/json")
+	if c.username != "" || c.password != "" {
```

Comment:
```suggestion
	if c.username != "" && c.password != "" {
```

shouldn't we use &&?

Reply:
I also contemplated that. Apparently, the description of how a URL should look like allows empty passwords - or empty users as well. With `&&` we would limit the code to accept URLs with both username and password filled. This way, we allow empty passwords.

Reply:
Thanks for explaining.